docs/eom_base.rst

```diff
@@ -0,0 +1,372 @@
+potential energy  of the whole system is going to be determined. ::
+
+    >>> KE = 0.5 * (m1*u1**2 + m2*u2**2)
+    >>> PE = symbols("PE")
+    >>> out_eqns = {"kinetic energy": KE, PE: 0.5*k1*x1**2 + 0.5*k2*(x2-x1)**2}
+
+The `eombase` instances are now ready to be initialized. To showcase possible
+input combinations, all three of the equations of motion forms will be used. ::
+
+    >>> eom1 = eombase.EOM(coordinates, RHS, speeds=speeds, 
+    ...                    output_eqns=out_eqns)
+    >>> eom2 = eombase.EOM(states, f_full, mass_matrix=mm_full,
+    ...                    num_coordinates=2)
+    >>> eom3 = eombase.EOM(states, f, mass_matrix=mm, coordinate_derivatives=G)
+
+There are some potential problems that can occur during the initialization of
+the eombase.EOM class. First, the class assumes if it recieves
+coordinate_derivatives that the input equations of motion are in form [3]. This
+means it will produce an error if a mass matrix is not recieved. Similarly if a
+mass matrix is recieved and coordinate_derivatives are not specified, the class
+is assuming that the equations of motion have been put in in form [2]. If the
+equations of motion are input in form [1], the class will not have a notion of
+the mass matrix or the forcing vector and as such will return errors if there
+is an attempt to access these attributes. ::
+
+    >>> eom4 = eombase.EOM(states, f, coordinate_derivatives=G)
+    SyntaxError: Need to specify a mass matrix for eom form [3]
+    >>> eom1.mass_matrix
+    AttributeError: Mass matrix is not specified for eom form [1]
+    >>> eom1.mass_matrix_full
+    AttributeError: Mass matrix full is not specified for eom form [1]
+    >>> eom1.forcing
+    AttributeError: Forcing vector is not specified for eom form [1]
+    >>> eom1.forcing_full
+    AttributeError: Full forcing vector is not specified for eom form [1]
+
+The system instance will now be initialized and set up to perform simulation. ::
+
+    >>> sys = System(eom1)
+    >>> sys.times = linspace(0, 10, num=100)
+    >>> sys.constants = {m1: 10.0, c1: 10.0, k1: 10.0, m2: 5.0, c2: 5.0, k2:
+    ...                  5.0}
+    >>> sys.initial_conditions = {x1: 1.0, x2: 0.0, u1: 0.0, u2: 0.0}
+
+Now the actual system simulation is ready to be run. ::
+
+    >>> out = sys.integrate()
+    >>> out
+    array([[1.0, 0.0, 0.0, 0.0], [...], ...])
+
+The user specified output equations should be able to be determined now that the
+time simulation of the states is complete. These equations are calculated by
+sys.output_eqns and so if there is an attempt to see the results first an error
+will be returned. ::
+
+    >>> eom.output_eqns_results
+    ValueError: The output equations first need to be calculated
+    >>> sys.output_eqns()
+    {"kinetic energy": array([[0.0], [...], ...]),
+     PE: array([[5.0], [...], ...])}
+    >>> eom1.output_eqns
+    {"kinetic energy": KE, PE: 0.5*k1*x1**2 + 0.5*k2*(x2-x1)**2}
+    >>> eom1.output_eqns_results
```

Comment:
I do not quite understand the 2d array form you have described. Since this is more related to pydy's System object now I'm going to move forward with SymbolicSystem in SymPy.